src/v/coproc/tests/utils/event_publisher.cc

```diff
@@ -117,6 +117,9 @@ ss::future<> event_publisher::create_coproc_internal_topic() {
 
 ss::future<std::vector<kafka::produce_response::partition>>
 event_publisher::publish_events(model::record_batch_reader reader) {
+    /// TODO: our kafka client doesn't support producing compressed batches,
```

Comment:
this should be easy to add no? we just compress it w/ zstd? i.e.: 

the client debounces each write (call to produce) ... when the debounce triggers (say 10ms), we create a batch, add headers like (panda-proxy: <user-agent>) and then compress? so i assume we have most of the infra, for that.

Reply:
Yes, I think compressing the batch after the debounce is easy, but we need to wire up the configuration to allow the user to specify the compression.

Reply:
@BenPope  - by default could be zstd (can be an enum: none .... snappy, etc, like kafka and then `default` which allows us to choose for the users)